### PR TITLE
Limit event msg size & event number per operation

### PR DESCRIPTION
**Pull request by sydhds**

### CODE REVIEW

File: massa-execution-exports/src/event_store.rs [hunk 1]
```diff
@@ -55,48 +55,61 @@ impl EventStore {
...
+
+    /// Get events iterator optionally filtered by given EventFilter
+    pub fn get_filtered_sc_output_events_iter<'b, 'a: 'b>(
+        &'a self,
+        filter: &'b EventFilter,
+    ) -> impl Iterator<Item = &'a SCOutputEvent> + 'b {
+        // Note on lifetimes:
+        // 'a -> is the lifetime for self -> because the iterator returns items from self
+        // 'b -> is the lifetime for filter -> because the returning iterator captures filter
+        // and we have lifetime 'a > 'b because filter can live less than self
```
Comment (bilboquet) [inline massa-execution-exports/src/event_store.rs:71]: do you mean "can't live"?

File: massa-execution-worker/src/interface_impl.rs [hunk 2]
```diff
@@ -1150,6 +1152,24 @@ impl Interface for InterfaceImpl {
 
         let mut context = context_guard!(self);
         let event = context.event_create(data, false);
```
Comment (bilboquet) [inline massa-execution-worker/src/interface_impl.rs:1154]: maybe move this line just after
```rust 
 if event_per_op >= self.config.max_event_per_operation {
    bail!("Too many event for this operation");
}
```
(only create the event if it's possible)

**Review by bilboquet — APPROVED** [review]
2 minor comments
it's looking good to me

File: massa-execution-exports/src/event_store.rs (hunk 1 above)
Comment (sydhds) [reply massa-execution-exports/src/event_store.rs:71]: No here lifetime 'b has the following bound: 'a: 'b. So it means that lifetime 'b can live less than lifetime 'a

File: massa-execution-worker/src/interface_impl.rs (hunk 2 above)
Comment (sydhds) [reply massa-execution-worker/src/interface_impl.rs]: Fixed

**Review by bilboquet — APPROVED** [review]

**Review by bilboquet — APPROVED** [review]

**Pull request merged**